src/libstd/primitive_docs.rs

```diff
@@ -482,8 +482,8 @@ mod prim_pointer { }
 /// an array. Indeed, this provides most of the API for working with arrays.
 /// Slices have a dynamic size and do not coerce to arrays.
 ///
-/// There is no way to move elements out of an array. See [`mem::replace`][replace]
-/// for an alternative.
+/// You can move elements out of an array with a slice pattern. If you want
```

Comment:
An example of this would be good.

Reply:
Is this example appropriate? (from The Edition Guide)

```
fn main() {
    greet(&[]);
    greet(&["Alan"]);
    greet(&["Joan", "Roa"]);
}

fn greet(people: &[&str]) {
    match people {
        [] => println!("Bummer, there's no one here :("),
        [only_one] => println!("Hey, there {}! You seem to be alone.", only_one),
        _ => println!("Hey everyone, we seem to be {} here today.", people.len()),
    }
}
```

Reply:
Yes =P but I'm biased since I wrote that example... ;)

Reply:
Oh wait... I got too excited; that one doesn't actually move out of the array because it's not an array... There's an example of that in the edition guide also.

Reply:
Hmm, what about this?

```
let arr = ["John".to_string(), "Roa".to_string()];

assert_eq!("contains two people", match arr {
    [_, _] => "contains two people",
    _ => "contains others",
});
```

Reply:
The `_ => ...` arm is unreachable there because the first pattern is irrefutable. I would do something like:

```rust
    fn move_away(_: String) { /* Do interesting things. */}

    let [john, roa] = ["John".to_string(), "Roa".to_string()];
    move_away(john);
    move_away(roa);
```

This clearly illustrates to the reader that `john` is indeed moved into `move_away`.

Reply:
Thank you for the advice! I added your example.